### PR TITLE
Implement advanced online players chart

### PR DESCRIPTION
**Pull request by KevinDaGame**
## What type of PR is this? (check all applicable)

- [x] Feature
- [ ] Refactor
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Add the bstats online players drilldown pie chart

## QA Instructions, Screenshots, Recordings
Start a server and get n players online

## Added/updated tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
  have not been included_
- [ ] I need help with writing tests
- [ ] Not needed

### CODE REVIEW

File: VoxelSniperSpigot/src/main/java/com/github/kevindagame/voxelsniper/SpigotVoxelSniper.java [hunk 1]
```diff
@@ -112,8 +114,8 @@ public void onEnable() {
         metrics.addCustomChart(new SimplePie("worldguard_integration", () -> WorldGuardIntegration.Companion.getEnabled() ? "enabled" : "disabled"));
         metrics.addCustomChart(new SimplePie("plotsquared_integration", () -> PlotSquaredIntegration.Companion.getEnabled() ? "enabled" : "disabled"));
         metrics.addCustomChart(new SingleLineChart("total_brush_uses_in_last_30_minutes", BrushUsageCounter::getTotalBrushUses));
-//        metrics.addCustomChart(new Metrics.MultiLineChart("uses_per_brush", BrushUsageCounter::getUsagePerBrush));
         metrics.addCustomChart(new SingleLineChart("total_snipers", BrushUsersCounter.Companion::getTotalBrushUses));
+        metrics.addCustomChart(new DrilldownPie("server_size_category_counter", () -> ServerSizeCategoryCounter.INSTANCE.getData(getServer().getOnlinePlayers().size())));
```
Comment (Lennart99) [inline VoxelSniperSpigot/src/main/java/com/github/kevindagame/voxelsniper/SpigotVoxelSniper.java:118]: What is the difference between a SimplePie and DrilldownPie? Why do we need the DrilldownPie here? (It seems to be more complex)

File: VoxelSniperSpigot/src/main/java/com/github/kevindagame/voxelsniper/integration/bstats/ServerSizeCategoryCounter.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+    )
+    fun getData(playerCount: Int): Map<String, Map<String, Int>> {
+        val map = mutableMapOf<String, Map<String, Int>>()
+        val entries = mutableMapOf<String, Int>()
+        val category = categories.find { playerCount in it.min..it.max }
+        entries[playerCount.toString()] = 1
+        map[category!!.label] = entries
+
+        return map
```
Comment (Lennart99) [inline VoxelSniperSpigot/src/main/java/com/github/kevindagame/voxelsniper/integration/bstats/ServerSizeCategoryCounter.kt:24]: Could you add a little bit of a comment on what is happening here? I see you are returning a map of maps here, but it is not fully clear to me.

**Review by Lennart99 — APPROVED** [review]
meant to click approve lol, see comments

File: VoxelSniperSpigot/src/main/java/com/github/kevindagame/voxelsniper/SpigotVoxelSniper.java (hunk 1 above)
Comment (KevinDaGame) [reply VoxelSniperSpigot/src/main/java/com/github/kevindagame/voxelsniper/SpigotVoxelSniper.java:118]: DrillDownPie lets us have subValues in categories. So we can quickly see an estimate median server size, but we can also see the individual exact numbers when you click on a group (see the java version chart)

File: VoxelSniperSpigot/src/main/java/com/github/kevindagame/voxelsniper/integration/bstats/ServerSizeCategoryCounter.kt (hunk 2 above)
Comment (Lennart99) [reply VoxelSniperSpigot/src/main/java/com/github/kevindagame/voxelsniper/integration/bstats/ServerSizeCategoryCounter.kt:31]: so the structure would than be like this?
{%category%: {%count%: 1} }

File: VoxelSniperSpigot/src/main/java/com/github/kevindagame/voxelsniper/SpigotVoxelSniper.java (hunk 1 above)
Comment (Lennart99) [reply VoxelSniperSpigot/src/main/java/com/github/kevindagame/voxelsniper/SpigotVoxelSniper.java:118]: ah ok

File: VoxelSniperSpigot/src/main/java/com/github/kevindagame/voxelsniper/integration/bstats/ServerSizeCategoryCounter.kt (hunk 2 above)
Comment (KevinDaGame) [reply VoxelSniperSpigot/src/main/java/com/github/kevindagame/voxelsniper/integration/bstats/ServerSizeCategoryCounter.kt:31]: Correct

**Pull request merged**